frontend/src/pages/room/[id].tsx

```diff
@@ -20,27 +23,43 @@ const GET_LATEST_PROGRAMMING_LANGUAGE_EVENT = "programming_language:get"
 const UPDATE_PROGRAMMING_LANGUAGE_EVENT = "programming_language:update"
 const RECEIVE_PROGRAMMING_LANGUAGE_EVENT = "programming_language:receive"
 
-export default function CodeRoom({ id, question }: PageProps) {
-
+export default function CodeRoom({ id, questionId }: PageProps) {
   const [isDomLoaded, setIsDomLoaded] = useState(false)
+  const [isQuestionLoaded, setIsQuestionLoaded] = useState(false)
   const [programmingLanguage, setProgrammingLanguage] = useState("python")
+  const [question, setQuestion] = useState<QuestionObject>({} as QuestionObject);
+
+  const token = useJwt();
   const [codeFromEditor, setCodeFromEditor] = useState("")
 
   function onProgrammingLanguageChange(language: string) {
     setProgrammingLanguage(language)
     socket.emit(UPDATE_PROGRAMMING_LANGUAGE_EVENT, { language: language, room: id })
   }
 
+  async function getQuestion() {
```

Comment:
I think it's better to fetch the question in getServerSideProps too! Just that to call `fetchQuestion` in `services/questions.ts`, we currently have to pass in a token, so I had to use the `useToken` hook in client before fetching the question. But since there's no check for token to get a question in our backend, do you think it'll work if I just remove the need for tokens when fetching questions in the frontend to make it simpler?

Reply:
Sounds good! Let's try that